pbjson-build/src/generator/message.rs

```diff
@@ -753,151 +753,207 @@ fn write_deserialize_field<W: Write>(
         json_name
     )?;
     writeln!(writer, "{}}}", Indent(indent + 1))?;
-    write!(writer, "{}{}__ = Some(", Indent(indent + 1), field_name)?;
+    write!(writer, "{}{}__ = ", Indent(indent + 1), field_name)?;
 
-    if let Some(one_of) = one_of {
-        write!(
-            writer,
-            "{}::{}(",
-            resolver.rust_type(&one_of.path),
-            field.rust_type_name()
-        )?;
-    }
-
-    match &field.field_type {
-        FieldType::Scalar(scalar) => {
-            write_encode_scalar_field(indent + 1, *scalar, field.field_modifier, writer)?;
-        }
-        FieldType::Enum(path) => match field.field_modifier {
-            FieldModifier::Repeated => {
-                write!(
-                    writer,
-                    "map.next_value::<Vec<{}>>()?.into_iter().map(|x| x as i32).collect()",
-                    resolver.rust_type(path)
-                )?;
-            }
-            _ => {
+    match one_of {
+        Some(one_of) => match &field.field_type {
+            FieldType::Scalar(s) => match override_deserializer(*s) {
```

Comment:
Fields within a oneof can't have labels, and so we don't need to worry about repeated, etc...